hstrat/dataframe/_surface_test_drive.py

```diff
@@ -0,0 +1,116 @@
+    hstrat surface annotations corresponding to the phylogeny tips, "as-if"
+    they had evolved according to the provided phylogeny history.
+
+    Parameters
+    ----------
+    df : pl.DataFrame
+        The input DataFrame containing alife standard phylogeny with required
+        columns, one row per taxon.
+
+        Note that the alife-standard `ancestor_list` column is not required.
+
+        Required schema:
+            - 'id' : pl.UInt64
+                - Taxon identifier.
+            - 'ancestor_id' : pl.UInt64
+                - Taxon identifier of ancestor.
+                - Own 'id' if root.
+
+        Optional schema:
+            - 'origin_time' : pl.UInt64
+                - Number of generations elapsed from ancestor.
+                - Determines branch lengths.
+                - Otherwise, all branches are assumed to be length 1.
+            - 'extant' : pl.Boolean
+                - Should an entry corresponding to this phylogeny taxon be
+                  included in the output population?
+                - Otherwise, all tips are considered extant and all inner nodes
+                  are not.
+            - Additional user-defined columns will be forwarded to the output
+              DataFrame.
+
+    dstream_algo : str
+        Name of downstream curation algorithm to use.
+
+    dstream_S : int
+        Capacity of annotation dstream buffer, in number of data items.
+
+    stratum_differentia_bit_width : int
+        The bit width of the generated differentia.
+
+    Returns
+    -------
+    pl.DataFrame
+        The output DataFrame containing generated hstrat surface annotations.
+
+        Required schema:
+            - 'data_hex' : pl.String
+                - Raw genome data, with serialized dstream buffer and counter.
+                - Represented as a hexadecimal string.
+            - 'dstream_algo' : pl.Categorical
+                - Name of downstream curation algorithm used.
+                - e.g., 'dstream.steady_algo'
+            - 'dstream_storage_bitoffset' : pl.UInt64
+                - Position of dstream buffer field in 'data_hex'.
+            - 'dstream_storage_bitwidth' : pl.UInt64
+                - Size of dstream buffer field in 'data_hex'.
+            - 'dstream_T_bitoffset' : pl.UInt64
+                - Position of dstream counter field in 'data_hex'.
+            - 'dstream_T_bitwidth' : pl.UInt64
+                - Size of dstream counter field in 'data_hex'.
+            - 'dstream_S' : pl.Uint32
+                - Capacity of dstream buffer, in number of data items.
+            - 'origin_time' : pl.UInt64
+                - Number of generations elapsed since the founding ancestor.
+
+        Additional user-defined columns will be forwarded from the input
+        DataFrame.
+    """
+    surfaces = hstrat.descend_template_phylogeny_alifestd(
+        alifestd_try_add_ancestor_list_col(
+            df.lazy().collect().to_pandas()
+        ),  # maybe don't need this
```

Comment:
Just took a look at the source, and it seems like `ancestor_id` column is needed but not necessarily the `ancestor_list` column